test/jdk/java/awt/Focus/AltTabEventsTest.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4524015
+ * @summary Tests that when user switches between windows using Alt-tab then the appropriate events are generated
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual AltTabEventsTest
+ */
+
+import java.awt.Button;
+import java.awt.Choice;
+import java.awt.Component;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+import java.awt.PopupMenu;
+import java.awt.event.ActionListener;
+import java.awt.event.ActionEvent;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+
+public class AltTabEventsTest {
+
+
+    private static final String INSTRUCTIONS = """
+           This test verifies that when user switches between windows using Alt-tab
+           key combination then appropriate window events are generated. Also, when
+           user interacts with Menu bar, Popup menu, Choice then no excessive window
+           event is generated.
+
+           After test started you will see Frame('Test for 4524015')-F1 with some
+           components and Frame('Another frame')-F2 with no components.
+           1. Make F1 active by clicking on it.
+           2. Press Alt-tab.
+           In the messqge dialog area you should see that
+           WINDOW_DEACTIVATED,WINDOW_LOST_FOCUS event were generated.
+           If you switched to F2 then also WINDOW_ACTIVATED, WINDOW_GAINED_FOCUS
+           were generated.
+           If no events were generated the test FAILED.
+           Repeat the 2) with different circumstances.
+
+           3. Make F1 active by clicking on it.
+           4. Click on Menu bar/Button 'popup'/Choice and select some item from
+           the list shown. If any of the window events appeared in the output then
+           the test FAILED.
+
+           else the test PASSED.""";
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("AltTabEventsTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 5)
+                .columns(35)
+                .testUI(AltTabEventsTest::createTestUI)
```

Comment:
can be changed to `new Test();` and thus leading to remove the `createTestUI()` method.

Reply:
ok

Reply:
Looks like you missed this.

Reply:
Sorry, it was a mistake on my side. Don't know why but updated test was not shown. It is updated now.